backends/firebase/index.js

```diff
@@ -219,7 +219,7 @@ export default class Firebase extends AuthBackend {
 				username: user.email,
 				name: user.displayName,
 				avatar: user.photoURL,
-				...user
+				raw: user,
```

Comment:
Why are we not using a `getUser()` in this backend?

Reply:
It has its own logic of handling auth state change — via the `onAuthStateChanged` event listener provided by the Firebase library. Technically, we don't make any requests to the auth server ourselves.

The Firebase backend is a special backend, you know. 😅

Reply:
Hm. Can we implement a `getUser()` then and simply not send a request? Otherwise plugins etc may call it, and they'll just get the parent class version.

Reply:
Not sure I’m following. Do you mean adding the `getUser()` method to the `Firebase` class, which overrides the one provided by the parent `AuthBackend` class?

I wonder, what if we could pass the `info` object as an optional parameter to the `getUser()` method and send a request to `userCall` only if `info` wasn’t passed?

```js
info ??= await this.request(...toArray(Class.userCall));
```

We would get a way to provide user data, like we can (programmatically) provide an access token, for backends, which don’t perform separate API calls to get user data but obtain them somehow differently. E.g., Firebase gets it by listening to the corresponding event.

Reply:
If you don't need the declarative API that makes `getUser()` simply work from the parent, you don't have to use it. But you *do* need to have *a* `getUser()` function defined, otherwise anything calling `backend.getUser()` will get the parent `getUser()` which will not work. I don't know how the Firebase backend gets user data, but you can use that within your `getUser()` implementation.

Reply:
Now I get it. Thank you. I will add `getUser()`. I should do it _after_ we merge this PR, right?